File: backend/meetmind_ai_backend/app/Models/Transcript.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Transcript extends Model
{
    protected $fillable = ['meeting_id', 'audio_file_id', 'text', 'language'];

    public function meeting(): BelongsTo
    {
        return $this->belongsTo(Meeting::class);
    }

    public function audioFile(): BelongsTo
    {
        return $this->belongsTo(AudioFile::class);
    }

    public function summary(): HasOne
    {
        return $this->hasOne(Summary::class);
    }

    public function taskCandidates(): HasMany
    {
        return $this->hasMany(TaskCandidate::class);
    }
}

File: lib/features/transcript/domain/entities/transcript.dart
import 'package:equatable/equatable.dart';

class Transcript extends Equatable {
  const Transcript({
    required this.id,
    required this.meetingId,
    required this.text,
    required this.createdAt,
    this.language,
  });

  final String id;
  final String meetingId;
  final String text;
  final String? language;
  final DateTime createdAt;

  @override
  List<Object?> get props => [id, meetingId, text, language, createdAt];
}
